src/index.ts

```diff
@@ -548,7 +548,8 @@ const createLogger = <Y extends string>(config?: configLoggerType) => {
     extend: (extension: string) => loggerType;
   };
 
-  const mergedConfig = { ...defaultLogger, ...config };
+  // Merge all non undefined values
+  const mergedConfig = { ...defaultLogger, ...JSON.parse(JSON.stringify(config)) };
```

Comment:
This can't work because functions are forward in the config, and function can't be parsed as json... This need to be reverted, it break everything

Reply:
fixed in v 5.2.2 (#108)